frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+class Root extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      tasks: null,
+      taskName: '',
+      taskDescription: '',
+      childVisible: false,
+      submitBar: false,
+      completedState: 0
+    };
+  }
+
+  componentDidMount() {
+    fetch('/task')
+      .then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
```

Comment:
when you're using fat arrow functions, you don't need so many brackets. Plus, fat arrows autoreturn unless you use `{ }`. So, you could write this function like so:

```javascript
fetch('/task')
  .then(response => response.json())
  .then(({ tasks }) => this.setState({ tasks }));
```

I'd also recommend pulling out a helper function, since you'll be repeating the `response.json()` part a lot. You could DRY this up like so:

```javascript
// request.js
// You'll probably want to introduce some error handling as well.
const request = (...args) => fetch(...args).then(response => response.json());
```

Now, you can simply:

```javascript
request('/tasks').then(({ tasks }) => this.setState({ tasks }));
```